Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Parameters/SpeckleBlockDefinitionWrapper.cs

```diff
@@ -20,30 +19,45 @@ public class SpeckleBlockDefinitionWrapper : SpeckleWrapper
 {
   public InstanceDefinitionProxy InstanceDefinitionProxy { get; set; }
 
-  ///<remarks>
-  /// `InstanceDefinitionProxy` wraps `Base` just like `SpeckleCollectionWrapper` and `SpeckleMaterialWrapper`
-  /// </remarks>
-  public override Base Base
+  public override required Base Base
   {
     get => InstanceDefinitionProxy;
     set
     {
-      if (value is not InstanceDefinitionProxy instanceDefinitionProxy)
+      if (value is not InstanceDefinitionProxy def)
       {
         throw new ArgumentException("Cannot create block definition wrapper from a non-InstanceDefinitionProxy Base");
       }
+      InstanceDefinitionProxy = def;
+    }
+  }
+
+  private List<SpeckleObjectWrapper> _objects = new();
 
-      InstanceDefinitionProxy = instanceDefinitionProxy;
+  public List<SpeckleObjectWrapper> Objects
+  {
+    get => _objects;
+    set
+    {
+      ValidateObjects(value);
+      _objects = value;
     }
   }
 
-  /// <summary>
-  /// Represents the objects contained within the block definition
-  /// </summary>
-  /// <remarks>
-  /// Objects can contain geometry, Speckle objects, Speckle instances
-  /// </remarks>
-  public List<SpeckleObjectWrapper> Objects { get; set; } = new(); // TODO: This isn't handling instances! Maybe List<SpeckleWrapper> but this wouldn't exclude Material etc. which isn't wanted.
+  private static void ValidateObjects(List<SpeckleObjectWrapper> objects)
+  {
```

Comment:
tbh I would introduce an empty "ISpeckleAtomicObject" interface to help with this, and eg in collections. You can see how I'm using a "ISpeckleProperty" interface in the nested props pr to do something similar

Reply:
Noted. As part of this pr's scope? Or should we clean these types of things up when merging `bjorn/cnx-1625-blocks-in-gh8` into `dev`?